### DISCUSSION TITLE
Kedro extension for PyCharm

### DISCUSSION THREAD

**Discussion by kzych-inpost** [kedro-org/kedro#4187] [Idea]
After the announcement of `Kedro VS Code Extension 0.2.2` I was wondering if there would be also a plugin for PyCharm.
I believe that many people use also this IDE so the support would be much appreciated. 

What do you think?

**Comment by astrojuanlu** [comment]
Thanks for opening this @kzych-inpost!

We investigated a bit some months ago:

>> https://blog.jetbrains.com/platform/2023/07/lsp-for-plugin-developers/
>
> I did some research today and found that in theory Pycharm support LSPs, but it's only limited to paid users which is disappointing.

_Originally posted by @noklam in https://github.com/kedro-org/kedro/issues/2821#issuecomment-2070424371_

Do you know what's the current status of LSP in PyCharm?

**Reply by kzych-inpost** [reply]
No idea about that :/ 

> It's only limited to paid users which is disappointing.

I am using PyCharm Professsional - don't you think that most of the Kedro users do the same?

**Reply by astrojuanlu** [reply]
> I am using PyCharm Professsional - don't you think that most of the Kedro users do the same?

The truth is that we have no idea, but it would be a good question to ask in a Kedro User survey 👍🏼

**Comment by noklam** [comment]
Update on this, with the new release of PyCharm they have open up the LSP API for all PyCharm user with 2025.1+ version. So the LSP API is no longer a blocker.

https://platform.jetbrains.com/t/update-on-lsp-and-template-language-apis-in-pycharm/1453

**Reply by astrojuanlu** [reply]
Looks like the conditions to start working on this are now met!

**Comment by noklam** [comment]
Are people still interested in this? or has everyone move to the Cursor / Copilot /Claude Code party already?

**Comment by noklam** [comment]
![lsp-pycharm](https://github.com/user-attachments/assets/c8aa8ab3-3f82-44bc-9774-feae0adb0f64) I may have a pycharm plugin working 👀

**Reply by rxm7706** [reply]
@noklam - Wow! this is something a lot of people would love to use - any plans / eta to releasing to the world!

**Reply by noklam** [reply]
@rxm7706 I don't have a timeline yet. The LSP is fully functional now and the remaining work is the kedro project discovery and some logistic of releasing the plugin.

Cc @rashidakanchwala

**Reply by noklam** [reply]
Some update, unfortunately this is not the priority of the team. Personally I do not have the time to properly open source this  by myself.

I will share the source code as is, this is a functional plugin with some changes that needed to be done:
- I copy the exising lsp server from the vscode-kedro repo, in theory they can shared the same codebase but it's not a separate package today.
- there are some hardcoded path inside the code that point to my local spaceflights directory for testing. This need some extra love to make sure the project discovery (i.e. for Pycharm to properly locate where the kedro project is) is correct.

Repo: https://github.com/noklam/pycharm-kedro/

If there are people who are interested to release the plugin on pycharm properly, I am happy to chat.
